### PR TITLE
Sensu Go cluster configuration guide update

### LINKED ISSUE #794: Update Sensu 2.0 clustering document to only use configuration on disk
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)

## Expected Behavior
Clustering guide provides exact steps to get a working cluster or when other options are provided (in this case config file vs command line arguments), those are called out as needed or not needed depending on your situtation.

## Current Behavior
The page:

https://docs.sensu.io/sensu-core/2.0/guides/clustering/#peer-communication-authentication-with-https-client-certificates

Shows starting the sensu-backend with command line options, when previously it had gone over setting up the configuration in `/etc/sensu/backend.xml`. Configuration shouldn't be needed in both places and should be called out.

## Possible Solution
Update the document as described above.

### PR DESCRIPTION
**Pull request by apaskulin**
## Description
Closes #794

## Motivation and Context
Current clustering guide has a mix of using on disk configuration and service configuration flags. Need to make it more clear when to use each option.

This PR will also include some overall updates to the guide to make it more clear to follow for new Sensu Go users.

Remaining:
- [x] Change external etcd commands to configuration on disk.
- [x] Add missing commands to make this guide more complete (start/restart commands, assumptions on already created/installed backends/agents).
- [ ] Find command to validate backend peer SSL certs.
- [x] Run through guide to check for configuration mistakes.

## Review Instructions
With @rgeniesse talking in the first person:

- [ ] Please check for overall feel. There's enough changes here that could be done better, I just need _someone_ else to look at this after 2 days.
- [ ] You can use the following skunkworks [vagrant box](https://github.com/sensu/skunkworks/tree/master/sensu-go-cluster-ssl) if you want to see a working SSL install based on this doc. *NOTE: the current state of the skunkworks code is to use the same cert for `etcd-cert-file` and `etcd-peer-cert-file`. The document generates a client cert with an IP and sensu/etcd complains:
```
Nov 21 02:46:20 backend-1 sensu-backend[28021]: WARNING: 2018/11/21 02:46:20 Failed to dial 10.0.0.1:2379: connection error: desc = "transport: authentication handshake failed: x509: cannot validate certificate for 10.0.0.1 because it doesn't contain any IP SANs"; please retry.
```
I'll have to talk with eng about it though to know the best update here.

Also note, that there are sections of this document I don't full understand, namely the sections of `Client-to-server transport security with HTTPS` and `Client-to-server authentication with HTTPS client certificates`. I _think_ I do, but not totally sure and the 2nd item above will help lead to answers as both these sections need more descriptions on their use cases.

### CODE REVIEW

**Comment by apaskulin** [comment]
To do: Add a note about setting up your cluster with flags

**Comment by rgeniesse** [comment]
@nikkiki, as you were the original creator of this guide, could I get a review from you when you have have time?

Happy to go over any of the changes and my reasoning to back them up, :-).

I was generally aiming for a more complete guide for someone setting up a cluster for the first time.

I am out all next week, but I'll be here on Friday. If you think you won't be able to get to it on short notice, I can type out some more notes about my changes. You can find some under the review instructions in this PR.

Thanks!

File: content/sensu-go/5.0/guides/clustering.md [hunk 1]
```diff
@@ -63,22 +108,22 @@ Get cluster health status and etcd alarm information.
 {{< highlight shell >}}
 $ sensuctl cluster health
          ID            Name      Error   Healthy  
- ────────────────── ─────────── ─────── ───────── 
-  a32e8f613b529ad4   backend-0           true     
-  c3d9f4b8d0dd1ac9   backend-1    wat    false     
-  c8f63ae435a5e6bf   backend-2           true
+ ────────────────── ─────────── ─────── ─────────
+  a32e8f613b529ad4   backend-1           true
+  c3d9f4b8d0dd1ac9   backend-2    wat    false
```
Comment (cwjohnston) [inline content/sensu-go/5.0/guides/clustering.md]: Is `wat` a real error a user might encounter in this output?

Comment (nikkictl) [reply content/sensu-go/5.0/guides/clustering.md]: Unfortunately, no. `dial tcp 10.0.0.2:2379: connect: connection refused` might be a more descriptive error.

**Comment by nikkictl** [comment]
We recently added support for `--etcd-advertise-client-urls` in https://github.com/sensu/sensu-go/pull/2375. We'll want to add this flag to our examples. In the use cases described, the advertise client urls would be the same as the backend IP address:
```
etcd-advertise-client-urls: "https://10.0.0.1:2379"
```

File: content/sensu-go/5.0/guides/clustering.md [hunk 2]
```diff
@@ -18,43 +18,88 @@ A Sensu Cluster is a group of [at least 3][1] sensu-backend nodes, each connecte
...
+_NOTE: This backend configuration assumes you have setup and installed the sensu-backend on all the nodes you will use in your cluster. You can use our [installation and configuration guide][14] guide if you have not done so_
+
+**backend-1**
 
-Below is an example configuration snippet from `backend.yml` where `10.n.0.1` and `backend-n` correspond to the names and addresses of the etcd nodes in the cluster (ex. `"backend-0=https://10.0.0.1:2380,backend-1=https://10.1.0.1:2380..."`).
 {{< highlight shell >}}
 ##
-# store configuration
+# store configuration for backend-1/10.0.0.1
 ##
```
Comment (nikkictl) [inline content/sensu-go/5.0/guides/clustering.md:52]:
```suggestion
##
etcd-advertise-client-urls: "https://10.0.0.1:2379"
```

File: content/sensu-go/5.0/guides/clustering.md [hunk 3]
```diff
@@ -18,43 +18,88 @@ A Sensu Cluster is a group of [at least 3][1] sensu-backend nodes, each connecte
...
 {{< /highlight >}}
 
-Using this configuration file at the start up of each sensu-backend accordingly, you should have a highly available Sensu Cluster! You can verify its health and try other cluster management commands using [sensuctl][6].
+**backend-2**
 
-### docker-compose.yml
+{{< highlight shell >}}
+##
+# store configuration for backend-2/10.0.0.2
+##
```
Comment (nikkictl) [inline content/sensu-go/5.0/guides/clustering.md:68]:
```suggestion
##
etcd-advertise-client-urls: "https://10.0.0.2:2379"
```

File: content/sensu-go/5.0/guides/clustering.md [hunk 4]
```diff
@@ -18,43 +18,88 @@ A Sensu Cluster is a group of [at least 3][1] sensu-backend nodes, each connecte
...
+etcd-name: "backend-2"
+{{< /highlight >}}
 
-If you'd prefer to stand up your Sensu Cluster within docker containers, check out the sensu-go [docker configuration][7]. This configuration defines 3 sensu-backend containers and 3 sensu-agent containers.
+**backend-3**
+
+{{< highlight shell >}}
+##
+# store configuration for backend-3/10.0.0.3
+##
```
Comment (nikkictl) [inline content/sensu-go/5.0/guides/clustering.md:84]:
```suggestion
##
etcd-advertise-client-urls: "https://10.0.0.3:2379"
```

**Comment by apaskulin** [comment]
Thanks, @nikkiki! I incorporated your suggestions and added `etcd-advertise-client-urls` to the backend reference as well.

**Review by nikkictl — APPROVED** [review]
Excellent job! LGTM!

**Pull request merged**